src/extractor/guidance/roundabout_handler.cpp

```diff
@@ -66,8 +68,15 @@ detail::RoundaboutFlags RoundaboutHandler::getRoundaboutFlags(
     bool on_roundabout = in_edge_data.roundabout;
     bool can_enter_roundabout = false;
     bool can_exit_roundabout_separately = false;
-    for (const auto &road : intersection)
+
+    const bool lhs = profile_properties.left_hand_driving;
+    const int step = lhs ? -1 : 1;
+    const signed int_size = static_cast<signed>(intersection.size());
```

Comment:
Why all the `signed` here and below? Just use `int` for the `-1/1` and `std::size_t` to count things.